jsonknife.c

```diff
@@ -785,14 +810,20 @@ knife_extract_max_timestamptz(PG_FUNCTION_ARGS) {
 
 	DateAccumulator acc;
 	acc.minmax = max;
-	acc.acc = 0;
+	acc.acc = NULL;
 
 	reduce_paths(value, paths, &acc, reduce_timestamptz);
 
-	if (acc.acc != 0)
-		PG_RETURN_NUMERIC(acc.acc);
-	else
+	if (acc.acc == NULL) {
 		PG_RETURN_NULL();
+	} else {
+		Datum result = *acc.acc;
+		pfree(acc.acc);
```

Comment:
Suspicious: memory use after free. (4)

Reply:
move acc.acc to result (through *) and free memory. Seems ok.